Sources/NIO/SelectorGeneric.swift

```diff
@@ -0,0 +1,333 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOConcurrencyHelpers
+
+internal enum SelectorLifecycleState {
+    case open
+    case closing
+    case closed
+}
+
+internal extension Optional {
```

Comment:
please move all the access controls away from `extension` to the actual declarations. In this case `internal` is default anyway so you can just remove it.

Reply:
(didn't actually touch that, but fixed in e19e685)